onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc

```diff
@@ -133,14 +133,15 @@ Status EmbedLayerNorm<T>::Compute(OpKernelContext* context) const {
 
   // Calculate mask
   if (nullptr != mask) {
-    // TODO: Consider summing the values in the mask and measure performance.
     const int32_t* mask_data = mask->template Data<int32_t>();
     int32_t* mask_index_data = mask_index->template MutableData<int32_t>();
     for (int b = 0; b < batch_size; b++) {
-      mask_index_data[b] =
-          static_cast<int32_t>(std::count_if(mask_data + (static_cast<int64_t>(b) * sequence_length),
-                                             mask_data + (static_cast<int64_t>(b) * sequence_length) + sequence_length,
-                                             [](int v) { return v == 1; }));
+      int32_t cur_sum = 0;
+      const int32_t* cur_mask_data = mask_data + (static_cast<int64_t>(b) * sequence_length);
+      for (int s = 0; s < sequence_length; ++s) {
+        cur_sum += cur_mask_data[s];
```

Comment:
curious - you don't need to check the condition cur_mask_data[s] == 1? If the condition is not important, would std::accumulate do better?

Reply:
Good catch! The data should always be 1 here - but we don't guarantee it. I profiled the regular vector sum w/ adding in a if `(vector[i] == 1)` and the performance gap remains the same. I've updated to include this.

Reply:
Thanks! Our tests didn't catch this :-).